Finance LLM leaderboard (#2389)
huggingface/blog: leaderboard-finbench.md

### Individual Tasks

We use 40 tasks on this leaderboard, across these categories:  
- **Information Extraction(IE)**: NER, FiNER-ORD, FinRED, SC, CD, FNXL, FSRL  
- **Textual Analysis(TA)**: FPB, FiQA-SA, TSA, Headlines, FOMC, FinArg-ACC, FinArg-ARC, MultiFin, MA, MLESG  
- **Question Answering(QA)**: FinQA, TATQA, Regulations, ConvFinQA  
- **Text Generation(TG)**: ECTSum, EDTSum  
- **Risk Management(RM)**: German, Australian, LendingClub, ccf, ccfraud, polish, taiwan, ProtoSeguro, travelinsurance  
- **Forecasting(FO)**: BigData22, ACL18, CIKM18  
- **Decision-Making(DM)**: FinTrade  
- **Spanish**: MultiFin-ES, EFP ,EFPA ,FinanceES, TSA-Spanish

For a detailed explanation or each task, please read the next section

> #### Explaination of Tasks

Author: @osanseviero do you maybe know how to add a section in a dropdown field? The task explanations are very interesting but maybe too dense for a blog

Reviewer: Do you mean like this? https://github.com/huggingface/blog/blob/6f63a7065716ce2d4830c010a11b9239d6487921/matryoshka.md?plain=1#L168

Author: Yesssss :heart:

> ---

Reviewer 2: I don't see any leaderboard link in the blog post. Should it be somewhere in the intro?

Author: Adding it for ref: https://huggingface.co/spaces/TheFinAI/Open-Financial-LLM-Leaderboard

I'll ask the partners if they want an embed for ex

Reviewer 2: And I think adding a link somewhere early on would make sense, otherwise it's a blog announcing a leaderboard without linking to it

Author: Completely agree, just need to see where we put it :)

* **Comprehensive Financial Task Coverage:** The leaderboard evaluates models only on tasks that are directly relevant to finance. These tasks include **information extraction**, **sentiment analysis**, **credit risk scoring**, and **stock movement forecasting**, which are crucial for real-world financial decision-making.  
* **Real-World Financial Relevance:** The datasets used for the benchmarks represent real-world challenges faced by the finance industry. This ensures that models are actually assessed on their ability to handle complex financial data, making them suitable for industry applications.  
* **Focused Zero-Shot Evaluation:** The leaderboard employs a **zero-shot evaluation** method, testing models on unseen financial tasks without any prior fine-tuning. This approach reveals a model’s ability to generalize and perform well in financial contexts, such as predicting stock price movements or extracting entities from regulatory filings, without being explicitly trained on those tasks.

> ## **Key Features of the Open Financial LLM Leaderboard**

Reviewer: ```suggestion
## Key Features of the Open Financial LLM Leaderboard
```

* **Forecasting (FO):** One of the most critical applications in finance is the ability to forecast market movements. Tasks under this category evaluate a model’s ability to predict stock price movements or market trends based on historical data, news, and sentiment. These tasks are central to tasks like portfolio management and trading strategies.  
* **Risk Management (RM):** This category focuses on tasks that evaluate a model’s ability to predict and assess financial risks, such as **Credit Scoring**, **Fraud Detection**, and **Financial Distress Identification**. These tasks are fundamental for credit evaluation, risk management, and compliance purposes.  
* **Decision-Making (DM):** In finance, making informed decisions based on multiple inputs (e.g., market data, sentiment, and historical trends) is crucial. Decision-making tasks simulate complex financial decisions, such as **Mergers & Acquisitions** and **Stock Trading**, testing the model’s ability to handle multimodal inputs and offer actionable insights.

> #### Metrics

Reviewer: ```suggestion
### Metrics
```

maybe?

### Individual Tasks

We use 40 tasks on this leaderboard, across these categories:  
- **Information Extraction(IE)**: NER, FiNER-ORD, FinRED, SC, CD, FNXL, FSRL  
- **Textual Analysis(TA)**: FPB, FiQA-SA, TSA, Headlines, FOMC, FinArg-ACC, FinArg-ARC, MultiFin, MA, MLESG  
- **Question Answering(QA)**: FinQA, TATQA, Regulations, ConvFinQA  
- **Text Generation(TG)**: ECTSum, EDTSum  
- **Risk Management(RM)**: German, Australian, LendingClub, ccf, ccfraud, polish, taiwan, ProtoSeguro, travelinsurance  
- **Forecasting(FO)**: BigData22, ACL18, CIKM18  
- **Decision-Making(DM)**: FinTrade  
- **Spanish**: MultiFin-ES, EFP ,EFPA ,FinanceES, TSA-Spanish

 <details><summary><b>Click here for a short explanation of each task</b></summary> 
   
1. **FPB (Financial PhraseBank Sentiment Classification)**  
   **Description:** Sentiment analysis of phrases in financial news and reports, classifying into positive, negative, or neutral categories.  

>    **Metrics:** Accuracy, F1-Score								Source: [https://huggingface.co/datasets/ChanceFocus/en-fpb](https://huggingface.co/datasets/ChanceFocus/en-fpb)  

Reviewer: Can we maybe add the source links to the title to make this a bit lighter?

Also, did you explore using a table to present this info?

Author: I personally like reading 40 paragraphs more than a table with 40 lines, but let's see what the authors think :)

* **Model Size:** Ranges from \~1.5 billion to 70+ billion parameters.

### Viewing Results in the Task Table

Once you've selected your tasks, the results will populate in the **task table** (see image). This table provides detailed metrics for each model across the tasks you’ve chosen. The performance of each model is displayed under columns labeled **Average IE**, **Average TA**, **Average QA**, and so on, corresponding to the tasks you selected.

### Submitting a Model for Evaluation

If you have a new model that you’d like to evaluate on the leaderboard, the **submission section** allows you to upload your model file. You’ll need to provide:

* **Model name**  
* **Revision commit**  
* **Model type**  
* **Precision**  
* **Weight type**

After uploading your model, the leaderboard will **automatically start evaluating** it across the selected tasks, providing real-time feedback on its performance.    

## Current Best Models and Surprising Results

Throughout the evaluation process on the Open FinLLM Leaderboard, several models have demonstrated exceptional capabilities across various financial tasks. 

As of the latest evaluation:

> - **Best model**: GPT-4 and Llama3.1 have consistently outperformed other models in many tasks, showing high accuracy and robustness in interpreting financial sentiment.  

Reviewer: ```suggestion
- **Best model**: GPT-4 and Llama 3.1 have consistently outperformed other models in many tasks, showing high accuracy and robustness in interpreting financial sentiment.  
```

> - **Surprising Results**: The **Forecasting(FO)** task, focused on stock movement predictions, showed that smaller models, such as **Llama3.1-7b, internlm-7b**,often outperformed larger models, for example Llama3.1-70b, in terms of accuracy and MCC. This suggests that model size does not necessarily correlate with better performance in financial forecasting, especially in tasks where real-time market data and nuanced sentiment analysis are critical. These results highlight the importance of evaluating models based on task-specific performance rather than relying solely on size or general-purpose benchmarks.

Reviewer: Links to models could be nice here.

> - **Surprising Results**: The **Forecasting(FO)** task, focused on stock movement predictions, showed that smaller models, such as **Llama3.1-7b, internlm-7b**,often outperformed larger models, for example Llama3.1-70b, in terms of accuracy and MCC. This suggests that model size does not necessarily correlate with better performance in financial forecasting, especially in tasks where real-time market data and nuanced sentiment analysis are critical. These results highlight the importance of evaluating models based on task-specific performance rather than relying solely on size or general-purpose benchmarks.

Reviewer: ```suggestion
- **Surprising Results**: The **Forecasting(FO)** task, focused on stock movement predictions, showed that smaller models, such as **Llama-3.1-7b, internlm-7b**,often outperformed larger models, for example Llama-3.1-70b, in terms of accuracy and MCC. This suggests that model size does not necessarily correlate with better performance in financial forecasting, especially in tasks where real-time market data and nuanced sentiment analysis are critical. These results highlight the importance of evaluating models based on task-specific performance rather than relying solely on size or general-purpose benchmarks.
```

> ## **Acknowledgments**

Reviewer: ```suggestion
## Acknowledgments
```
